spec/project_config_spec.rb

```diff
@@ -792,4 +792,168 @@
       expect(feature_flag).to eq(config_body['featureFlags'][0])
     end
   end
+
+  # Only those log messages have been asserted, which are directly logged in these methods.
+  # Messages that are logged in some internal function calls, are asserted in their respective function test cases. 
+  describe 'get_forced_variation' do
```

Comment:
@mikeng13 We already have 3 invalid cases covered here. The method 'present' works on local and travis but fails on e2e testing. It's not a default method of Ruby, and is a part of active support extension. Anyway, I have replaced all of them with has_key which looks safe